File: solutions_11-20/Euler_16.py
# what is the sum of the digits 2**1000 ? 

import timeit


start = timeit.default_timer()

def euler_16():
    return sum([int(i) for i in str(2**1000)])

print "Answer: %s" % euler_16()
stop = timeit.default_timer()
print "Time: %f" % (stop - start)

File: Java/java_solutions_11-20/Euler_16.java
/*
/   Project Euler problem #16 Power digit sum
/
/   What is the sum of the digits in 2 to the power of 1000?
/
*/

import java.math.BigInteger;


public class Euler_16
{
    public static void main(String[] args)
    {
        int exponentLimit = 1000;
        int base = 2;


        BigInteger powerOf = new BigInteger("1");
        BigInteger baseExp = new BigInteger("2");

        for (int i = 1; i <= exponentLimit; i++)
            powerOf = powerOf.multiply(baseExp);

        String baseExpStr = powerOf.toString();

        long expSum = 0;
        for (int i = 0; i < baseExpStr.length(); i++)
            expSum += Integer.parseInt(baseExpStr.substring(i, i + 1));

        System.out.println("Answer: " + expSum);
    }
}
